Create_lab_1

Monstazim opened the pull request.

Monstazim committed:
Create_lab_1

Monstazim committed:
Создал проект

Monstazim committed:
Удалил лишнее

BoychikFromAsia commented on Lab1/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,26 @@
+﻿#include <iostream>
+#include <stdio.h>;
+struct Stack
```
в лаб 0 это не требуется

GMtimur commented on 2sem/common/stack.h:
```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <iostream>
```
Стек ничего не должен знать о системе ввода/вывода. Фактически этим инклюдом сократили область применения вашей библиотеки только тем окружением, где есть iostream

ichepkov2401 replied:
@Monstazim после того, как исправили тред или в рамках обсуждения пришли к решению ничего не менять, резолвите треды

Monstazim replied:
исправил

BoychikFromAsia commented on Lab1/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,26 @@
+﻿#include <iostream>
+#include <stdio.h>;
+struct Stack
```
в лаб 0 это не требуется

ichepkov2401 replied:
Ну исходя из номер ветки и названия PR - это первая лабораторная, а не нулевая

Monstazim committed:
Написал первую лабу

1. Реализовал структуру данных стек Stack<T> по методу LIFO
2. Реализовал следующие методы:
a. void push(T) - добавляет элемент на вершину стека
b. T pop() - извлекает элемент с вершины стека и возвращает его значение
c. T peek() - возвращает элемент находящийся на вершине стека
d. int count() - возвращает количество элементов в коллекции
e. void clear() - удаляет все элементы из коллекции

Monstazim committed:
Исправление ошибок

1. Исправлена ошибка "cerr: необъявленный идентификатор", с помощью использования оператора throw
2. Исправлены названия переменных

Monstazim committed:
Исправление ошибок №2

Исправлена ошибка "cerr: необъявленный идентификатор", с помощью использования оператора throw в T pop()

Monstazim committed:
удалил ненужные файлы

eggorrus commented on 2sem/common/stack.h:
```diff
@@ -0,0 +1,94 @@
+#pragma once
+#include <stdexcept>
+using namespace std;
+/// <summary>
```
Битая кодировка

eggorrus commented on 2sem/common/stack.h:
```diff
@@ -0,0 +1,94 @@
+using namespace std;
+/// <summary>
+/// ���������� �������
+/// Template Declaration
```
Комментарии нужно оставлять либо только на русском, либо только на английском, не нужно дублировать

eggorrus commented on 2sem/common/stack.h:
```diff
@@ -0,0 +1,94 @@
+/// ���������� �������
+/// Template Declaration
+/// </summary>
+/// /// 
```
Пустой тег

eggorrus commented on 2sem/common/stack.h:
```diff
@@ -0,0 +1,94 @@
+	/// �������, ������� ��������� ������� �� ������� �����
+	/// A function that adds an element to the top of the stack
+	/// </summary>
+	void push(const T& value)
```
Здесь не нужно передавать в метод ссылку

eggorrus commented on 2sem/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,30 @@
+{
+    setlocale(LC_ALL, "");
+    int newValue;
+    Stack<int> Lab1Stack;
```
Неудачное имя для переменной, лучше использовать stack или myStack, а Lab1Stack вообще не по стайлгайду

eggorrus commented on 2sem/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,30 @@
+        cin >> newValue;
+    }
+
+    int currentValue;
```
Эта переменная не нужна, лучше обойтись без неё (речь про sizeOfStack)

eggorrus replied:
для переменной currentValue слишком большая область видимости

eggorrus commented on 2sem/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,30 @@
+    int currentValue;
+    int sizeOfStack = Lab1Stack.count();
+    cout << "Положительные числа последовательности в обратном порядке:" << endl;
+    for (int i{ 0 }; i < sizeOfStack; i++)
```
Лучше использовать цикл while(stack.count()>0)

eggorrus commented on 2sem/Lab1/Lab1.cpp:
```diff
@@ -0,0 +1,30 @@
+    Stack<int> Lab1Stack;
+    cout << "Введите последовательность чисел" << endl;
+    cin >> newValue;
+    while (newValue != 0)
```
нужно добавить логические разделители между частями кода, чтобы он не выглядел монолитным полотном
